### PR TITLE
Test automatic Git checkouts on the web UI

### PR DESCRIPTION
**Pull request by Martchus**
Related ticket: https://progress.opensuse.org/issues/161771

### CODE REVIEW

File: tests/osautoinst/test_running.pm [hunk 1]
```diff
@@ -16,6 +16,8 @@ sub run ($self) {
         assert_script_run q{retry -s 30 -r 5 -- sh -c 'openqa-cli api jobs state=running state=done | ack --passthru --color "running|done"'}, 300;
     }
     save_screenshot;
+    # verify that the test distribution is checked out
+    assert_script_run q{retry -s 10 -r 6 -e -- sh -c 'test -f /var/lib/openqa/share/tests/*/.git/config'}, 600 if get_var('OPENQA_FROM_BOOTSTRAP');
```
Comment (okurz) [inline tests/osautoinst/test_running.pm:20]:
```suggestion
    assert_script_run q{retry -s 10 -r 6 -e -- sh -c 'test -f /var/lib/openqa/share/tests/*/.git/config'}, 600 if get_var('OPENQA_FROM_BOOTSTRAP'), fail_message => 'the test distribution should be checked out by openQA automatically';
```

**Review by okurz — APPROVED** [review]

**Review by kalikiana — APPROVED** [review]

**Comment by Martchus** [comment]
I fixed the scenario of the manual installation. It worked in a local test run. Let's see whether the installation via the bootstrap script works (as it might not be available yet).

**Comment by okurz** [comment]
https://openqa.opensuse.org/tests/4266606 seems to say no

**Comment by Martchus** [comment]
The boostrap package installed in this test is still at 1501646c92d50d30d976d8c6c58770d010a05063. So I guess we'll have to wait until the package is updated in Tumbleweed (as we don't add our devel repo in https://openqa.opensuse.org/tests/4266606#step/openqa_webui/1).

**Comment by Martchus** [comment]
I know. I restarted the Jenkins job manually after it previously failed (on 5xx errors), see https://progress.opensuse.org/issues/161771#note-11.

**Comment by Martchus** [comment]
Waiting for https://github.com/os-autoinst/openQA/pull/5709 to reach TW.

In the scenario https://openqa.opensuse.org/tests/4279626#step/test_running/6 we can already see that it is working. I also tested the mentioned PR in the SUT in a developer mode session.

**Comment by okurz** [comment]
https://github.com/os-autoinst/os-autoinst-distri-openQA/pull/183 should help you as it allows to add any repo to the test by configuration also for the case of using openQA-bootstrap so that you could easily test changes to the bootstrap script before needing to wait any further for PRs or SRs to be accepted

**Comment by Martchus** [comment]
I cloned the job via
```
openqa-clone-custom-git-refspec https://github.com/os-autoinst/os-autoinst-distri-openQA/pull/177 https://openqa.opensuse.org/tests/4285633 ADD_OPENQA_REPO=1 NEEDLES_DIR=https://github.com/os-autoinst/os-autoinst-needles-openQA.git
```
and it passed with the new check: https://openqa.opensuse.org/tests/4285751#step/test_running/6

(It was required to set `NEEDLES_DIR` to avoid `isotovideo died: Can't init needles from /var/lib/openqa/pool/28/os-autoinst-needles-openQA at /usr/lib/os-autoinst/needle.pm line 288.` but that's another problem.)

**Comment by okurz** [comment]
New Tumbleweed snapshot was released: https://lists.opensuse.org/archives/list/factory@lists.opensuse.org/thread/KA47VU6TEU4WL3DTN6WN2CFQXW62GH6Q/
retriggering openQA CI test.

**Pull request merged**